fix: General Ledger translation issues

rtdany10 opened the pull request:
Options of group_by filter are translated. Assertions were made against English sentences and hence would always return false in cases where English isn't the system language.

Solution: don't translate options server side. Only displayed labels are translated client side. 

closes #27277

rtdany10 commented:
@ankush unrelated test failing. Can we merge this?

rtdany10 commented:
@ankush

ankush reviewed (changes requested):
A better solution would be to remove translations from business logic altogether.

The options are anyways translated, so it's not required in the code. 

Example: `value` attribute contains actual value while display label can be whatever translate function returns. 

![Screenshot 2021-09-07 at 5 22 32 PM](https://user-images.githubusercontent.com/9079960/132340250-7b160d27-3c39-49ff-81fe-90b6fd9237bc.png)

ankush committed:
fix: remove translations from GL report options

Options need not be translated, their display label gets translated
client side.

ankush commented:
Oh well, removing all translation calls also removes it from `_messages` 🤦

ankush committed:
fix: make group by options translatable

ankush committed:
ci: semgrep rule for translated options in report
